### DISCUSSION TITLE
Incompatible with `immutable` Svelte compiler option

### DISCUSSION THREAD

**Discussion by imhul** [janosh/svelte-multiselect#195] [Q&A] [answered]
I have some bug with svelte-multiselect. Selected filters are not deleted by pressing the delete button. But in REPL is no bugs

REPL: https://svelte.dev/repl/2f17296c51504d2b82fe38c3f869aaba?version=3.55.1

video: https://drive.google.com/file/d/1E11B0vj93VMHY238QdvuD7bmBw7xo-B0/view?usp=sharing

**Comment by janosh** [comment]
Could you provide a repro? Either a REPL or StackBlitz would be great. You can click the readme badges to get you started:

[![REPL](https://img.shields.io/badge/Svelte-REPL-blue?label=Svelte)](https://svelte.dev/repl/a5a14b8f15d64cb083b567292480db05) [![Open in StackBlitz](https://img.shields.io/badge/StackBlitz-darkblue?logo=stackblitz)](https://stackblitz.com/github/janosh/svelte-multiselect)

**Reply by imhul** [reply]
still relevant

**Reply by nosovk** [reply]
The problem in 
```
	compilerOptions: {
		immutable: true
	},
```

when it's true, then we get behavior from video

**Reply by janosh** [reply]
@imhul The Google Drive link is not public, can't view the video. But like you said, looks like the REPL works fine.

**Reply by nosovk** [reply]
https://user-images.githubusercontent.com/1398808/212748725-04fb93b2-7497-4bab-b36d-070490acb94c.mp4

**Reply by janosh** [reply]
@nosovk Thanks for posting the video here. I see the problem. And you say the problem disappears when setting `compilerOptions.immutable: false`? I've never used that option myself. Is it generally compatible with Svelte? Svelte components tend to make use of mutability a lot...

**Reply by nosovk** [reply]
The default value for that option is false. 

Temporary fix is:
```
on:remove={(e) => selected = [...selected]}
```

It's not about mutability, but about change detection:
https://svelte.dev/repl/4301a81c2ead4d85af7673bf9f1a16e8?version=3.16.4
check example above for better explanation.

**Reply by nosovk** [reply]
Another article with explanation why immutable is needed:
https://sveltesociety.dev/recipes/svelte-language-fundamentals/options-immutable

**Comment by janosh** [comment, marked as answer]
@imhul Ah, thanks for clarifying. That should be an easy fix.

**Reply by janosh** [reply]
Should be fixed by #196. No imminent release planned unless this is time critical?

**Reply by imhul** [reply]
> Not a critical. Thank you very much for your help
